services/UserAuthService.ts

```diff
@@ -42,6 +42,27 @@ export default class UserAuthService {
     });
   }
 
+  public async modifyEmail(user: UserModel, proposedEmail: string): Promise<UserModel> {
+    const updatedUser = await this.transactions.readWrite(async (txn) => {
+      const userRepository = Repositories.user(txn);
+
+      proposedEmail = proposedEmail.toLowerCase();
+
+      // ensure that the proposed email isn't used already
+      const foundUser = await userRepository.findByEmail(proposedEmail);
+      if (foundUser !== undefined) {
+        throw new BadRequestError('Email already in use');
+      }
```

Comment:
nit: we can shorten this a bit and use the same logic we use in `UserAuthService::registerUser` for checking duplicates:
```ts
const emailAlreadyUsed = !!(await userRepository.findByEmail(registration.email));
if (emailAlreadyUsed) throw new BadRequestError('Email already in use');
```